kasa/deviceconfig.py

```diff
@@ -117,12 +129,11 @@ class DeviceConfig:
     host: str
     timeout: Optional[int] = DEFAULT_TIMEOUT
     port_override: Optional[int] = None
-    credentials: Credentials = field(
-        default_factory=lambda: Credentials(username="", password="")
-    )
+    credentials: Optional[Credentials] = None
+    credentials_hash: Optional[str] = None
```

Comment:
We should start slowly documenting these (using the format like shown in https://github.com/python-kasa/python-kasa/blob/master/kasa/smartbulb.py#L94-L97 which produces https://python-kasa.readthedocs.io/en/stable/smartbulb.html#kasa.TurnOnBehaviors), but that can wait for another PR, just something to keep in mind wrt #600.